internal/v1/handler_blueprints.go

```diff
@@ -84,6 +75,69 @@ func (bb *BlueprintBody) RedactPasswords() {
 	}
 }
 
+// Merges Password or SshKey from other User struct to this User struct if it is not set
+func (u *User) MergeExisting(other User) {
+	if u.Password == nil {
+		u.Password = other.Password
+	}
+	if u.SshKey == nil {
+		u.SshKey = other.SshKey
+	}
+}
+
+// User must have name and non-empty password or ssh key
+func (u *User) hasValidValues() error {
```

Comment:
```suggestion
func (u *User) isValid() error {
```

or even more Go native:

```suggestion
func (u *User) Valid() error {
```

There is no point not exporting this function I think, it is a safe one to export.

Reply:
exported and renamed